src/ng/directive/select.js

```diff
@@ -116,9 +120,26 @@ var SelectController =
 
   self.registerOption = function(optionScope, optionElement, optionAttrs, interpolateValueFn, interpolateTextFn) {
 
-    if (interpolateValueFn) {
+    if (interpolateValueFn === true) {
```

Comment:
I am not keen on using `true` as a special flag here.
Perhaps we could just add yet another parameter to the `registerOption` method?